src/InterestPayout.Common/Application/RecurringPayoutsScheduler.cs

```diff
@@ -37,43 +37,13 @@ public class RecurringPayoutsScheduler : IRecurringPayoutsScheduler
             _payoutConfigService = payoutConfigService;
             _idGenerator = idGenerator;
         }
-        
-        public async Task Init()
-        {
-            _logger.LogInformation("Initialization of scheduled recurring messages is started");
 
-            var configs = _payoutConfigService.GetAll();
-            await CancelSchedulesNotPresentInConfig(configs);
-            
-            foreach (var config in configs)
-            {
-                await using var unitOfWork = await _unitOfWorkManager.Begin($"SettingUpSchedules:{config.AssetId}:{DateTimeOffset.UtcNow.Ticks}");
-                try
-                {
-                    await HandleScheduleInitForConfigEntry(config, unitOfWork.PayoutSchedules);
-                    await unitOfWork.Commit();
-                }
-                catch (Exception e)
-                {
-                    _logger.LogError(e, "An error occurred during initialization of payout schedule config {@context}", new
-                    {
-                        config
-                    });
-                    await TrySafeCancelScheduleAfterError(config, e);
-                    throw;
-                }
-            }
-            
-            _logger.LogInformation("Initialization of scheduled recurring messages has successfully finished");
-        }
-
-        private async Task CancelSchedulesNotPresentInConfig(IReadOnlyCollection<PayoutConfig> configs)
+        public async Task Remove(ISet<string> assetIds)
         {
-            var assets = configs.Select(x => x.AssetId).ToHashSet();
-            await using var unitOfWork = await _unitOfWorkManager.Begin($"CancellingSchedulesNotPresentInCfg:{DateTimeOffset.UtcNow.Ticks}");
+            await using var unitOfWork = await _unitOfWorkManager.Begin($"RemovingScheduleEntries:{DateTimeOffset.UtcNow.Ticks}");
```

Comment:
I assume that this method is triggered by an API invocation, so externally passed indemptency ID could be used instead of current time